ToDoApp/AddItemViewController/AddItemViewModel.swift

```diff
@@ -16,10 +17,13 @@ class AddItemViewModel {
         self.onOpenListVC?()
     }
     
-    func addValue(value: String?) {
-        guard let value = value else { return }
-        taskManager.data.append(value)
-        shouldOpenListVC()
+    
+    func addValue(productName: String?, productCount: String?) {
+        guard let name = productName, let count = productCount else {return}
+        if productName != "" && productCount != "" {
```

Comment:
you can checn it at 1 logic iteration
guard let name = productName, name != "" ...

or

if let name = productName, name != "" ....